### ISSUE TITLE
Ingress issue when trying to do a firmware update

### ISSUE THREAD

**Issue by razvanws** [hassio-addons/addon-zwavejs2mqtt#51]
# Problem/Motivation

Firmware update doesn't work using the addon-zwavejs2mqtt.

## Expected behavior

Firmware update should work properly using addon-zwavejs2mqtt.

## Actual behavior

Whenever "Begin firmware update" node action is selected, after you select the firmware file and the file has to be sent to zwavejs2mqtt, the following error can be seen in logs:

`2021/02/19 16:37:38 [error] 418#418: *2061 recv() failed (104: Connection reset by peer) while reading response header from upstream, client: 172.30.32.2, server: a0d7b954-zwavejs2mqtt, request: "POST /socket.io/?EIO=4&transport=polling&t=NUwkOdc&sid=XCa3GYmkbC7UkAYzAAAI HTTP/1.1", upstream: "http://127.0.0.1:44920/socket.io/?EIO=4&transport=polling&t=NUwkOdc&sid=XCa3GYmkbC7UkAYzAAAI", host: "10.10.111.4:8124", referrer: "https://10.10.111.4:8124/api/hassio_ingress/HF_cM1Ef8ZQhxN1KonodqGhXcHda5DE6tScRqD_P11o/control-panel"
`
If you expose the container port and use it directly (eg: 10.10.111.4:44920), the firmware update works without any problem, so it's not a problem with zwavejs2mqtt itself.

## Steps to reproduce

1. Open zwavejs2mqtt web ui.
2. Select a device.
3. On Node actions selectbox, choose "Begin firmware update"
4. Click ok and select the firmware file.
5. Check the addon logs, you will see the error desribed above.

## Proposed changes

Most probably, there's something missing or wrong in the ingress configuration (nginx) since accessing the container directly works as expected.

As a workaround, you can connect to zwavejs2mqtt container, install SSH on it, then use SSH to expose the port to the host and access zwavejs2mqtt directly (where 10.10.111.4 is the host IP address and 172.30.33.2 is the container IP address):
Eg. ssh -NfL 10.10.111.4:44920:0.0.0.0:44920 root@172.30.33.2

**Comment by razvanws** [comment]
It still doesn't work but it seems that no one is reading those reports.

**Comment by jwk4946** [comment]
I get this error in my logs constantly even without trying to manually update firmware even with latest release

**Comment by jasoncodes** [comment]
Still the same error with the following versions:

core: 2021.6.6
supervisor: 2021.06.6
addon-zwavejs2mqtt: 0.21.0
zwavejs2mqtt: 5.1.0
zwave-js: 7.9.0

**Comment by jasoncodes** [comment]
Still the same error with the following versions:

core: 2021.7.4
supervisor: 2021.06.8
addon-zwavejs2mqtt: 0.23.2
zwavejs2mqtt: 5.4.2
zwave-js: 8.0.5

**Comment by jwk4946** [comment]
I believe this issue still exists and has not been closed

**Comment by jasoncodes** [comment]
Yup. Just tested this again with the following versions:

core: 2021.8.8
supervisor: 2021.08.1
addon-zwavejs2mqtt: 0.24.1
zwavejs2mqtt: 5.5.1
zwave-js: 8.2.1

**Comment by jasoncodes** [comment]
Still an issue with the following versions:

supervisor: 2021.09.6
core: 2021.9.7
addon-zwavejs2mqtt: 0.25.1
zwavejs2mqtt: 5.6.2
zwave-js: 8.3.1

**Comment by curvingedge** [comment]
I am also having issues with zwavejs2mqtt and the zwavejs integration. I don't really know what is going on but my network used to be rock solid and now its not. I get timeouts and slow response from z-wave devices. I was able to perform an update with zwjs2mqtt but it took over 30 minutes. I could see the log would run and then pause for awhile the run and pause. This started after a HAOS update maybe two times ago. I suspect a usb problem but don't know what I need to prove it.

**Comment by mundschenk-at** [comment]
Is there anything special in your HassOS config? I've recently updated the FW on two devices and it was quite a finicky process on the Z-Wave side (device needed to be very neer to the controller), but no problem on the web interface side.

**Comment by jasoncodes** [comment]
Confirmed to be still an issue with the following versions:

supervisor: 2021.10.8
core: 2021.11.1
addon-zwavejs2mqtt: 0.28.0
zwavejs2mqtt: 5.10.1
zwave-js: 8.6.1

FYI this will likely be my last update to keep the stale bot at bay.

---

Here’s my one-liner to port-forward the container running on my Home Assistant Supervised host called `hass` and make it available locally as http://127.0.0.1:44920/:

```
ssh -N -L "44920:$(ssh hass docker inspect -f '{{range.NetworkSettings.Networks}}{{.IPAddress}}{{end}}' addon_a0d7b954_zwavejs2mqtt):44920" hass
```

**Comment by zetlen** [comment]
Still happening as of:

supervisor: 2021.12.2
core: 2021.12.7
addon-zwavejs2mqtt: 0.32.2
zwavejs2mqtt: 6.2.0
zwave-js: 8.9.1
